### PR TITLE
Updated Discord api version

### LINKED ISSUE #960: Upgrade to gateway version `10`
https://github.com/discord/discord-api-docs/discussions/4510

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
I have updated the discord api version to 10

closes #960

TODO:
- [x] tweak identifying payload
- [x] add all the changes for the new version.

### CODE REVIEW

**Review by KILLEliteMaste — COMMENTED** [review]
1. Your change does not work, it should be `10`
2. You have not done any changes mentioned in the changelog of the v10. The link is in the issue.

**Comment by RealYusufIsmail** [comment]
oh ok

File: javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java [hunk 1]
```diff
@@ -1564,6 +1571,9 @@ public CompletableFuture<Collection<Ban>> getBans(Integer limit, Long after) {
         ).setUrlParameters(getIdAsString());
 
         if (limit != null) {
+            if (limit < 0 || limit > 1000) {
+                request.addQueryParameter("limit", "1000");
+            }
```
Comment (KILLEliteMaste) [inline javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java:1576]: Why?

File: javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java [hunk 2]
```diff
@@ -1555,6 +1555,13 @@ public CompletableFuture<Ban> requestBan(long userId) {
                 .execute(result -> new BanImpl(this, result.getJsonBody()));
     }
 
+    /**
+     * This method is used the get the {@link Member} who are banned from the server.
+     * @param limit how many bans should be returned at most. Must be within [0, 1000]. If null, it will default to 1000
+     * @param after should be a snowflake to only take bans of users with IDs higher
+     *              than this parameter into account; can be null
+     * @return a list of bans
+     */
```
Comment (KILLEliteMaste) [inline javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java:1564]: Why did you add / edit the Javadoc in the implementation and not in the interface available in the API module?

File: javacord-api/src/main/java/org/javacord/api/entity/intent/Intent.java [hunk 3]
```diff
@@ -159,7 +159,26 @@ public enum Intent {
...
      */
-    DIRECT_MESSAGE_TYPING(14, false);
+    DIRECT_MESSAGE_TYPING(14, false),
+
+    /**
+     * MESSAGE_CONTENT is a special case as it doesn't represent individual events,
+     * but rather affects the data sent for most events that could contain message content fields
+     * (content, attachments, embeds, and components).
+     */
+    MESSAGE_CONTENT(15, false),
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/entity/intent/Intent.java:169]: This is a privileged intent, so it has to be true

**Review by KILLEliteMaste — CHANGES_REQUESTED** [review]
Since the attachments for messages are now removed when editing their attachments, this is a breaking change and the MessageUpdater needs to be updated properly to be able to use the new feature.
The changelog also states `/channels/<channel_id>/threads/active has been removed` which you did not

**Comment by KILLEliteMaste** [comment]
> Screenshot 2022-05-16 at 16 43 12
> 
> It is not there

Then we have this endpoint not implemented. I thought we support everything for threads but then it doesn't seem like that.

**Comment by RealYusufIsmail** [comment]
I will work on MessageUpdater  later today.

**Comment by RealYusufIsmail** [comment]
@KILLEliteMaste what would I change with the Message updater. Like do i need to change anything in MessageBuilderBase or some other class ?

**Comment by KILLEliteMaste** [comment]
> @KILLEliteMaste what would I change with the Message updater. Like do i need to change anything in MessageBuilderBase or some other class ?

I don't know, that's something you have to figure out. Since the MessageBuilderBase contains some functionality that's used by the MessageBuilder/Updater it may be possible that you need to change something even in the MessageBuilder to correctly adjust to changes in the MessageBuilderBase

**Comment by RealYusufIsmail** [comment]
Ok

**Comment by RealYusufIsmail** [comment]
Will start working on this today.

**Pull request closed without merging**